bundles/binding/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/SagerCasterBinding.java

```diff
@@ -0,0 +1,245 @@
+								eventPublisher.postUpdate(itemName, compassState);
+							}
+							State windTrend = new DecimalType(sagerWeatherCaster.getWindEvolution());
+							for (final String itemName : provider.getAllItemNames(CommandType.WINDTREND)) {
+								eventPublisher.postUpdate(itemName, windTrend);
+							}
+							postNewForecast(provider);
+						} else
+							logger.warn("Not enough historic data to study wind bearing evolution, wait a bit ...");
+					    break;
+					}
+					case SEALEVELPRESSURE : {
+						DecimalType newValue = (DecimalType) newState;
+						HistoricItem historicItem = PersistenceExtensions.historicState(item, earlier);
+						if (historicItem != null) {
+							DecimalType previousValue = (DecimalType) historicItem.getState();
+							sagerWeatherCaster.setPressure(newValue.doubleValue(), previousValue.doubleValue());
+							State pressTrend = new DecimalType(sagerWeatherCaster.getPressureTrend());
+							for (final String itemName : provider.getAllItemNames(CommandType.PRESSURETREND)) {
+								eventPublisher.postUpdate(itemName, pressTrend);
+							}
+							postNewForecast(provider);
+						} else
+							logger.warn("Not enough historic data to study pressure evolution, wait a bit ...");
+						break;
+					}
+					case CLOUDLEVEL :{
+						DecimalType newValue = (DecimalType) newState;
+						sagerWeatherCaster.setCloudLevel(newValue.doubleValue());
+						postNewForecast(provider);
+						break;
+					}
+					case RAINING : {
+						OnOffType newOnOffValue = (OnOffType) newState;
+						sagerWeatherCaster.setRaining(newOnOffValue.equals(OnOffType.ON));
+						postNewForecast(provider);
+						break;
+					}
+					case WINDSPEED : {
+						DecimalType newValue = (DecimalType) newState;
+						sagerWeatherCaster.setBeaufort(newValue.intValue());
+						postNewForecast(provider);
+						break;
+					}
+				}
+			}
+		}
+		
+	}
+
+	private void postNewForecast(SagerCasterBindingProvider provider) {
+		for (final String itemName : provider.getAllItemNames(CommandType.FORECAST)) {
+    		eventPublisher.postUpdate(itemName, new StringType(sagerWeatherCaster.getForecast()));
+	    }
+		for (final String itemName : provider.getAllItemNames(CommandType.VELOCITY)) {
+    		eventPublisher.postUpdate(itemName, new StringType(sagerWeatherCaster.getWindVelocity()));
+	    }
+		for (final String itemName : provider.getAllItemNames(CommandType.WINDFROM)) {
+    		eventPublisher.postUpdate(itemName, new DecimalType(sagerWeatherCaster.getWindDirection()));
+	    }
+		for (final String itemName : provider.getAllItemNames(CommandType.WINDTO)) {
+    		eventPublisher.postUpdate(itemName, new DecimalType(sagerWeatherCaster.getWindDirection2()));
+	    }
+	}
+
+	@Override
+	public void stateUpdated(Item item, State state) {
+		doCast(item,state);		
+	}
+	
+	@Override
+	public void stateChanged(Item item, State oldState, State newState) {
+		doCast(item,newState);
+	}
```

Comment:
rather than listening to stateChanged() events better listen to internalReceiveUpdate(). Implementing StateChangeListener can be omitted …